### PR TITLE
Add capability to add tools and miscellaneous resources

### LINKED ISSUE #67: Add capability to add tools and miscellaneous resources
**Description:**
In some cases, we need to ship tools like API Manager CLI, etc. We need to add the capability of copying such tools/resources to Ansible agents.

### PR DESCRIPTION
**Pull request by VimukthiPerera**
## Purpose
Resolves #67 

## Goals
Add capability to add tools and miscellaneous resources.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
Ansible 2.7.0
Ansible 2.8.0

### CODE REVIEW

File: dev/group_vars/apim-analytics.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ product_name: wso2am-analytics
 product_version: 2.6.0
 target: /mnt
 product_package_location: files
+misc_file_location: "{{ product_package_location }}/misc"
+misc_file_copy_location: "{{ target }}/misc"
```
Comment (Savidude) [inline dev/group_vars/apim-analytics.yml]: What if the user wants to copy the files to somewhere outside the `misc` directory? Shall we remove this and have the user set the `dest` in `misc_file_list` to anywhere they want?

File: roles/apim-gateway/tasks/main.yml [hunk 2]
```diff
@@ -70,6 +70,20 @@
         dest: "{{ item.dest }}"
       loop: "{{ config_files }}"
 
+    - name: Create miscellaneous file directory
```
Comment (Savidude) [inline roles/apim-gateway/tasks/main.yml]: Is this necessary?

File: README.md [hunk 3]
```diff
@@ -108,6 +114,8 @@ Copy the following files to `files/lib` directory.
 1. [MySQL Connector/J](https://dev.mysql.com/downloads/connector/j/5.1.html)
 2. [Amazon Corretto for Linux x64 JDK](https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html)
 
+Copy the miscellaneous files to `files/misc` directory.
```
Comment (Savidude) [inline README.md]: Shall we also mention that `misc_file_list` for the respective roles should be uncommented and the `src` and `dest` values should be set?

File: dev/group_vars/apim-analytics.yml (hunk 1 above)
Comment (VimukthiPerera) [reply dev/group_vars/apim-analytics.yml]: If the user wants to change the place where the files are copied, they can change the value assigned to `misc_file_copy_location`.

File: roles/apim-gateway/tasks/main.yml (hunk 2 above)
Comment (VimukthiPerera) [reply roles/apim-gateway/tasks/main.yml]: yes.

File: README.md (hunk 3 above)
Comment (VimukthiPerera) [reply README.md]: +1

Comment (VimukthiPerera) [reply README.md]: Fixed with https://github.com/wso2/ansible-apim/pull/70/commits/35cd23cf3403ed75f4c3a476f69d0ea68bb23781

File: roles/apim-gateway/tasks/main.yml (hunk 2 above)
Comment (Savidude) [reply roles/apim-gateway/tasks/main.yml]: There might be cases where different miscellaneous files are copied to locations outside `misc_file_copy_location`. I'm thinking that we could remove this and have the user create the relevant directories and set their own copy destinations.

File: dev/group_vars/apim-analytics.yml (hunk 1 above)
Comment (Savidude) [reply dev/group_vars/apim-analytics.yml]: Refer https://github.com/wso2/ansible-apim/pull/70#discussion_r309539646

**Pull request merged**